dialoptions.go

```diff
@@ -377,7 +382,15 @@ func WithInsecure() DialOption {
 // later release.
 func WithNoProxy() DialOption {
 	return newFuncDialOption(func(o *dialOptions) {
-		o.copts.UseProxy = false
```

Comment:
The godoc says the following:
```
This is ignored if WithDialer or WithContextDialer are used.
```
Since the proxy handling is now in the resolver, will the dialer always see the proxy's address? If so, this seems to be a behaviour change for users of custom dialers. Maybe we should skip proxy handling in the resolver if a custom dialer is being used to maintain backward compatability.